File: item-list/src/main/java/com/gurunars/item_list/ChangePersist.java
package com.gurunars.item_list;

import android.support.v7.widget.RecyclerView;

import java.util.List;

class ChangePersist<ItemType extends Item> implements Change<ItemType> {

    @Override
    public int apply(RecyclerView.Adapter adapter, Scroller scroller, List<ItemType> items,
                     int currentPosition) {
        // Do nothing
        return currentPosition;
    }

}

File: item-list/src/main/java/com/gurunars/item_list/ChangePersist.kt
package com.gurunars.item_list

import android.support.v7.widget.RecyclerView

internal class ChangePersist<ItemType : Item> : Change<ItemType> {

    override fun apply(adapter: RecyclerView.Adapter<*>, scroller: Scroller, items: MutableList<ItemType>,
                       currentPosition: Int): Int {
        // Do nothing
        return currentPosition
    }

}
